students/imx-lk/lesson03/mailroom_pt1.py

```diff
@@ -0,0 +1,156 @@
+    new_name=response_thankyou.title()
+    new_donor_list.append(new_name)
+
+    #Get the donation amount and append it to the new_donor_list
+    response_donation = int(input("\nEnter a donation amount(Ex: if $1,000 enter 1000): "))
+    new_donor_list.append(response_donation)
+
+    #append to donor_list so it can be added to the master list
+    donor_list.append(new_donor_list)
+
+    return new_donor_list
+
+
+def create_report(donor_list):
+#this list will capture the total, average, and number of gifts
+    list_total_avg = []
+
+#This for loop grabs the first list within donor_list
+    for donor_entry in donor_list:
+        #Need to clear out the total and average everytime or it adds up donations continuesly
+        total_donations= 0
+        average_donations = 0
+        #this temp list is to append my current items and then append it to a master list so I can create a list within a list
+        temporary_list = []
+        #get length to conpute the average and number of gifts
+        length_entry = (len(donor_entry)-1)
+        #accesing the first list  within donor_list I sliced and started at index 1. because I want to only access the donations amount
+
+        #sum all the donations
+        for donation in donor_entry[1:]:
+            total_donations +=donation
+
+        #calculate the average
+        average_donations = total_donations/length_entry
+        #append all the items to a temporary list
+        temporary_list.append(donor_entry[0])
+        temporary_list.append(total_donations)
+        temporary_list.append(length_entry)
+        temporary_list.append(average_donations)
+
+        #append the temporarly list to the master list to create a list within a list
+        list_total_avg.append(temporary_list)
+
+    print(end = '\n\n')
+    print('{:<20} {:^50} {:^5} {:>25}'.format('Donor Name', 'Total Given', 'Num Gifts', 'Average Gift'))
+    print('---------------------------------------------------------------------------------------------------------')
+    #for name in list_total_avg:
+        #print('{:<10} {:^45f} {:^5} {:>30f}'.format(*name))
+    for x in list_total_avg:
+        task=f'{x[0]:<20} {x[1]:^50.2f} {x[2]:} {x[3]:>30.2f}'
+        print(task)
+
+
+if __name__ == '__main__':
+    while True:
+        responses_available = ['1','2','q','Q']
+        response = input('Hello, What would you like to do: Send a Thank You (1), Create a Report(2) or quit(q): ')
+        while response not in responses_available:
+            response = input('No, enter one of the following: Send a Thank You (1), Create a Report(2) or quit(q): ')
+
+        #Call the function according to which option in the menu they picked
+        if response == '1':
+            thankyou_note(donor_list)
+        elif response == '2':
+            create_report(donor_list)
+        else:
+            print('\nGoodbye thank you for using the mailroom!')
+            break
+        continue
```

Comment:
A couple things missing here.  1) Report should be sorted by total given in descending order.  2) When adding new donations, it should add to the total for a donor, but it's currently creating duplicate donors.